crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -1200,16 +1200,11 @@ impl<'a> Visitor<'a> for Checker<'a> {
                             self.visit_non_type_definition(arg);
                         }
                         for arg in args {
-                            if let Expr::Dict(ast::ExprDict {
-                                keys,
-                                values,
-                                range: _,
-                            }) = arg
-                            {
-                                for key in keys.iter().flatten() {
-                                    self.visit_non_type_definition(key);
-                                }
-                                for value in values {
+                            if let Expr::Dict(ast::ExprDict { items, range: _ }) = arg {
+                                for ast::DictItem { key, value } in items {
```

Comment:
Same as for the visitor. The new implementatio changes the visitation order from keys, then values to: for each pair, `key` then `value`. Does this match pythons evaluation order?